### PR TITLE
Make diverses default directories more available

### PR DESCRIPTION
**Pull request by levitte**
## Add the possibility to display and use MODULESDIR

This adds the flag OPENSSL_MODULES_DIR for OpenSSL_version(), and the
flag '-m' for 'openssl version'.

##  Add a way for the application to get OpenSSL configuration data

OpenSSL_version(OPENSSL_DIR) gives you a nicely formatted string for
display, but if all you really want is the directory itself, you were
forced to parsed the string.

This introduces a new function to get diverse configuration data from
the library, OPENSSL_info().  This works the same way as
OpenSSL_version(), but has its own series of types, currently
including:

OPENSSL_INFO_CONFIG_DIR         returns OPENSSLDIR
OPENSSL_INFO_ENGINES_DIR        returns ENGINESDIR
OPENSSL_INFO_MODULES_DIR        returns MODULESDIR
OPENSSL_INFO_DSO_EXTENSION      returns DSO_EXTENSION

For scripting purposes, this also adds the command 'openssl info'.

### CODE REVIEW

**Comment by levitte** [comment]
This might for example be useful for @slontis' effort with FIPS module installation and testing.

**Comment by levitte** [comment]
Hold on, I'm going to rework this one a bit.

**Comment by levitte** [comment]
Rework done.

File: CHANGES [hunk 1]
```diff
@@ -9,6 +9,11 @@
 
  Changes between 1.1.1 and 3.0.0 [xx XXX xxxx]
 
+  *) Added OPENSSL_info() to get diverse built in OpenSSL data, such
```
Comment (slontis) [inline CHANGES]: Change to built-in otherwise it reads funny.

Comment (levitte) [reply CHANGES]: I'll never get the hang of when to and when not to use dashes...  Fixing

Comment (levitte) [reply CHANGES]: (and we use "built in" all over the place, it seems.  Still fixing my stuff here, though

Comment (slontis) [reply CHANGES]: The only other thing I can think of is the pesky windows backslashes, when building a path

Comment (mattcaswell) [reply CHANGES]: > I'll never get the hang of when to and when not to use dashes

Neither will I.

Comment (levitte) [reply CHANGES]: Ah, directory-file separator.  Sure, and that will fix the same problem for VMS.

Comment (levitte) [reply CHANGES]: Also, list separator character may be useful (we use LIST_SEPARATOR internally in a few places)

Comment (levitte) [reply CHANGES]: Okie, both added

**Comment by richsalz** [comment]
Use a dash, "built-in" when it is an adjective like "built-in data"

Less helpfully, do not use a dash when it is a verb, as in "built in the factory"

**Comment by levitte** [comment]
Rebased because util/libcrypto.num.

Final review, anyone?

**Comment by levitte** [comment]
Ping?

File: doc/man1/info.pod [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+[B<-configdir> | B<-c>]
+[B<-enginesdir> | B<-e>]
+[B<-modulesdir> | B<-m>]
+[B<-dsoext>]
+[B<-dirfilesep>]
+[B<-listsep]>
+
+=head1 DESCRIPTION
+
+This command is used to print out info information about OpenSSL.
```
Comment (mattcaswell) [inline doc/man1/info.pod]: info information?

**Review by mattcaswell — COMMENTED** [review]
I'm not really sure why we need a new command line app for this?

File: doc/man1/info.pod (hunk 2 above)
Comment (levitte) [reply doc/man1/info.pod]: Yes, exactly.  It's info about the information that you will eventually...

...  nah, kidding, will fix.

Comment (levitte) [reply doc/man1/info.pod]: I ended up adding a bit more

**Comment by levitte** [comment]
Ping?

**Review by paulidale — APPROVED** [review]
LGTM.

**Comment by paulidale** [comment]
After the conflicts are resolved.

**Comment by levitte** [comment]
Merged with ordinal fixup.

0109e030db9207a47e195b4c3a3b13e9017f0ed2 Add a way for the application to get OpenSSL configuration data
47ca8338358b01ef429a3801ce6173f7a0791674 Add the possibility to display and use MODULESDIR

**Pull request closed without merging**